Purchases/Networking/RCHTTPClient.m

```diff
@@ -122,37 +154,56 @@ - (void)      handleResponse:(NSURLResponse *)response
                        error:(NSError *)error
                      request:(NSMutableURLRequest *)request
            completionHandler:(RCHTTPClientResponseHandler)completionHandler
-beginNextRequestWhenFinished:(BOOL)beginNextRequestWhenFinished {
-    NSInteger statusCode = RC_NETWORK_CONNECT_TIMEOUT_ERROR;
+beginNextRequestWhenFinished:(BOOL)beginNextRequestWhenFinished
+                  retryBlock:(RetryRequestBlock)retryBlock
+                     retried:(BOOL)retried {
+    NSInteger statusCode = RCHTTPStatusCodesNetworkConnectTimeoutError;
     NSDictionary *responseObject = nil;
-
+    RCHTTPResponse *httpResponse = [[RCHTTPResponse alloc] initWithStatusCode:statusCode responseObject:responseObject];
     if (error == nil) {
         statusCode = ((NSHTTPURLResponse *) response).statusCode;
 
-        RCDebugLog(RCStrings.network.api_request_completed, request.HTTPMethod, request.URL.path, (long)statusCode);
+        RCDebugLog(RCStrings.network.api_request_completed, request.HTTPMethod, request.URL.path, (long) statusCode);
 
         NSError *jsonError;
-        responseObject = [NSJSONSerialization JSONObjectWithData:data
-                                                         options:0
-                                                           error:&jsonError];
-
+        if (statusCode == RCHTTPStatusCodesNotModifiedResponseCode && data.length == 0) {
```

Comment:
That's a good point, but that would be unexpected. 304s should always return "{}", but I think it would be safer to do it for every 304 so I think I am going to remove the check for data.length.